src/scripts/main.js

```diff
@@ -1,3 +1,332 @@
+
+// document.body.appendChild(form);
+
+const formHTML = `
+  <form class="new-employee-form">
+    <label>
+      Name: <input name="name" type="text" data-qa="name" required>
+    </label>
+    <label>
+      Position: <input name="position" type="text" data-qa="position">
+    </label>
+    <label>
+      Age: <input name="age" type="number" data-qa="age" required>
+    </label>
+    <label>
+      Salary: <input name="salary" type="number" data-qa="salary" required>
+    </label>
+    <label>
+      Office:
+      <select name="office" data-qa="office">
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <button type="submit">Save to table</button>
+  </form>
+`;
+
+document.body.insertAdjacentHTML('beforeend', formHTML);
+
+const form = document.querySelector('.new-employee-form');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  validateForm();
+});
+
+// function validateForm() {
+//   const nameEmployee = form.elements.name.value;
+//   const position = form.elements.position.value;
+//   const office = form.elements.office.value;
+//   const age = parseInt(form.elements.age.value, 10);
+//   const salary = parseFloat(form.elements.salary.value);
+
+//   if (nameEmployee.length < 4) {
+//     showNotification(
+//       'Error',
+//       'Name must be at least 4 characters long.',
+//       'error',
+//     );
+
+//     return;
+//   }
+
+//   if (!position) {
+//     showNotification('Error', 'Please choose a valid position.', 'error');
+
+//     return;
+//   }
+
+//   if (age < 18 || age > 90) {
+//     showNotification('Error', 'Age must be between 18 and 90.', 'error');
+
+//     return;
+//   }
+
+//   if (isNaN(salary) || salary <= 0) {
+//     showNotification('Error', 'Salary must be a positive number.', 'error');
+
+//     return;
+//   }
+
+//   addEmployeeToTable(nameEmployee, position, office, age, salary);
+//   showNotification('Success', 'Employee added successfully!', 'success');
+// }
+
+function validateForm() {
+  const nameEmployee = form.elements.name.value;
+  const position = form.elements.position.value;
+  const office = form.elements.office.value;
+  const age = parseInt(form.elements.age.value, 10);
+  const salary = parseFloat(form.elements.salary.value);
+
+  const validationError = validateEmployeeDetails(
+    nameEmployee,
+    position,
+    age,
+    salary,
+  );
+
+  if (validationError) {
+    showNotification('Error', validationError, 'error');
+
+    return;
+  }
+
+  addEmployeeToTable(nameEmployee, position, office, age, salary);
+  showNotification('Success', 'Employee added successfully!', 'success');
+}
+
+function validateEmployeeDetails(nameWorker, position, age, salary) {
```

Comment:
The function `validateEmployeeDetails` is returning a string in case of an error and `null` in case of success. This is not an intuitive API. Consider throwing an Error in case of a validation error and not returning anything in case of success. This way, the function can be used in a more intuitive way: if it throws, there was a validation error; if it doesn't throw, the validation was successful.